gdeep/search/hpo.py

```diff
@@ -306,14 +311,30 @@ def _objective(self, trial: BaseTrial, config: HPOConfig):
         # str(self.schedulers_param)
         # create a new model instance
         self.model = self._initialise_new_model(self.models_hyperparam)
-        self.pipe = Trainer(
-            self.model,
-            self.dataloaders,
-            self.loss_fn,
-            self.writer,
-            self.training_metric,
-            self.k_fold_class,
-        )
+        if self.regularize == True:
```

Comment:
@hkirvesl please add a unit test for the hpo (a very simple one) that tests this pathway

Reply:
Having to add `RegulariedTrainer` in parts of the code that does not have anything to do with regularization looks very fishy.

Reply:
done